extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/group/UserGroupService.java

```diff
@@ -87,9 +87,9 @@ private ExprNode getGroupSearchFilter() throws GuacamoleException {
         if (confService.getConfigurationBaseDN() != null)
             return new NotNode(new EqualityNode("objectClass","guacConfigGroup"));
```

Comment:
The filter should also be factored in right here - that is, if the user has configured the configuration DN, the configured search filter should, at the very least, be `AndNode`d together here with this `NotNode` to remove `guacConfigGroup` objects from the results.

Reply:
You say... Retrieve everything under `configurationbaseDN ` **NOT** containing `guacConfigGroup `right?

Reply:
Retrieve everything under `groupBaseDN` NOT containing `guacConfigGroup`.  Basically, the previous code made the assumption that, if you've configured a DN for configuration items you've also extended your LDAP schema to include the `guacConfigGroup` schema item, and that you do not want those being returned by the groups filter, you want them being returned in the config search.  That is still the case in the line mentioned above, but the new group filter needs to be and-ed with that logic.

Reply:
@necouchman do you find more convinient to replace "default" value of my proposed configuration item **ldap-group-search-filter** from `EqualityNode("objectClass","group")` to the original behaviour` PresenceNode("objectClass")`? 
In this way, if admin does not specify such filter then original behavior will be inplace. Otherwise, they will have the possibility of fine-grained control of which Groups shall be retrieved.

Reply:
My point, here, is that, as you have this implemented, when `ldap-config-base-dn` is set, the new `ldap-group-search-filter` property will be completely ignored, and you'll always end up with the following group filter:
`(!(objectClass=guacConfigGroup)))`

So, the lines above should be something like:
```
if (confService.getConfigurationBaseDN() != null)
    return new AndNode(new NotNode(new EqualityNode("objectClass","guacConfigGroup"), confService.getGroupSearchFilter());
```
Assuming the default group filter, this will result in an LDAP filter like this:
`(&(!(objectClass=guacConfigGroup))(objectClass=group))`